api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java

```diff
@@ -31,7 +53,137 @@
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(new Date(1950, -1, 2), result);
+    }
+    
+    @Test
+    public void testClearTimeComponentofNewDay(){
+    	Date date = new Date(2018, 7, 20);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(date, result);
+    }
 
+	@Test
+   	public void shouldSetupContext() {
+   		assertNotNull(Context.getService(ProviderManagementService.class));
+   	}
+    
+    @Test
+    public void checkWhenSupervisorListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = new Person();
+    	person.setId(2);
+    	Person supervisor = new Person();
+    	supervisor.setId(2);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(person);
+    	Person supervisorForPerson = providerManagementService.getSupervisorsForProvider(person).get(0);
+    	List<ProviderPersonRelationship> supervisors = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisors.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), supervisorForPerson.getId(),  relationship, relationshiptype));
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkWhenSupervisorListIsEmpty() throws PersonIsNotProviderException {
+    	Provider provider = new Provider();
+    	Person person = new Person();
+    	person.setId(2);
+    	provider.setPerson(person);
+    	Assert.assertTrue(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkForTheAccurateNumberOfSupervisorsAdded() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException, DateCannotBeInFutureException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = new Person();
+    	person.setId(2);
+    	Person supervisor = new Person();
+    	supervisor.setId(2);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(person);
+    	Person supervisorForPerson = providerManagementService.getSupervisorsForProvider(person).get(0);
+    	List<ProviderPersonRelationship> supervisors = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisors.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), supervisorForPerson.getId(),  relationship, relationshiptype));
```

Comment:
Again, I'm confused as to what the previous five lines are supposed to be doing?